### ISSUE TITLE
Basic VPC no longer works

### ISSUE THREAD

**Issue by jaxxstorm** [pulumi/pulumi-awsx#1131]
### What happened?

I have some fairly standard code for a VPC that worked in awsx v1. This fails in awsx v2 with:

```
File "/Users/lbriggs/src/github/jaxxstorm/pulumi-examples/python/aws/vpc/venv/lib/python3.11/site-packages/pulumi_aws/ec2/nat_gateway.py", line 615, in __init__
        NatGatewayArgs._configure(_setter, **kwargs)
      File "/Users/lbriggs/src/github/jaxxstorm/pulumi-examples/python/aws/vpc/venv/lib/python3.11/site-packages/pulumi_aws/ec2/nat_gateway.py", line 63, in _configure
        raise TypeError("Missing 'subnet_id' argument")
    TypeError: Missing 'subnet_id' argument
```

### Example

```
import pulumi
import pulumi_awsx as awsx

vpc = awsx.ec2.Vpc(
    "vpc",
    cidr_block="172.16.0.0/16",
    subnet_specs=[
        awsx.ec2.SubnetSpecArgs(
            type=awsx.ec2.SubnetType.PUBLIC,
            tags={
                "kubernetes.io/role/elb": "1",
            },
        ),
        awsx.ec2.SubnetSpecArgs(
            type=awsx.ec2.SubnetType.PRIVATE,
            tags={
                "kubernetes.io/role/internal-elb": "1",
            },
        ),
    ],
)
```

### Output of `pulumi about`

```
CLI          
Version      3.89.0
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
aws     6.6.1
awsx    2.0.3
docker  4.4.4
python  unknown

Host     
OS       darwin
Version  14.0
Arch     arm64

This project is written in python: executable='/opt/homebrew/bin/python3' version='3.11.4'

Current Stack: jaxxstorm/python-vpc/dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/jaxxstorm
User           jaxxstorm
Organizations  jaxxstorm, lbrlabs, team-ce, demo, pulumi
Token type     personal

Dependencies:
NAME         VERSION
pip          23.3.1
pulumi-awsx  2.0.3
setuptools   68.2.2
wheel        0.41.2

Pulumi locates its logs in /var/folders/rk/2c715ngd14jc4c7d3659m7pm0000gn/T/ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jaxxstorm** [comment]
Looks like https://github.com/pulumi/pulumi/issues/14418

**Comment by justinvp** [comment]
This actually isn't https://github.com/pulumi/pulumi/issues/14418. It's a different bug.

In this case, we're deserializing a resource reference from AWS. It's calling https://github.com/pulumi/pulumi-aws/blob/f9c925744b030b6318ab1aa842a9dee8c95c7e65/sdk/python/pulumi_aws/_utilities.py#L210

```python
    def construct(self, name: str, typ: str, urn: str) -> pulumi.Resource:
        class_name = self.mod_info['classes'].get(typ, None)

        if class_name is None:
            raise Exception(f"unknown resource type {typ}")

        TheClass = getattr(lazy_import(self.mod_info['fqn']), class_name)
        return TheClass(name, pulumi.ResourceOptions(urn=urn))
```

In this case, it's rehydrating a `NatGateway` resource, so the call looks like `NatGateway(name, pulumi.ResourceOptions(urn=urn))`.

When `pulumi.ResourceOptions(urn=urn)` it's not expected that any args are specified for the resource. The bug is that `_construct` isn't taking that into account. It's still erroring about missing expected argument, when it shouldn't because the `urn` resource option is set.

In any case, https://github.com/pulumi/pulumi/pull/14427 has reverted the change that introduced `_configure` in the first place, which was released in v3.91.1. Once `pulumi-aws` and `pulumi-awsx` are updated to use the v3.91.1 codegen and new versions are released, this shouldn't be an issue.

**Comment by mikhailshilkov** [comment]
This is fixed in 2.1.0 by upgrading to pu/pu 3.91.1
